routers/web/repo/repo.go

```diff
@@ -396,6 +396,23 @@ func RedirectDownload(ctx *context.Context) {
 			ServeAttachment(ctx, att.UUID)
 			return
 		}
+	} else if vTag == "latest" {
```

Comment:
I specified len(releases) == 0 so we don't serve the last one when multiple releases found with same tag

Reply:
TBH, the old code also seems related/suspicious. For example, the `GetReleasesByRepoIDAndNames` above: if there is no `GetReleasesByRepoIDAndNames(ctx, curRepo.ID, tagNames)`, it would `return err` early? Is there still a chance to execute your code?

Sorry that by reading the code, I can't really understand what it is dong or whether it is right.

If there are some tests or some other people could help, then I might be able to follow.

Reply:
I don't think it returns an error when nothing is found, I may need to check this.

Reply:
After reading the code again, I think I can understand the problem now. The old error-check code of `IsErrAttachmentNotExist` is wrong. So I removed it, then the code looks better.